### ISSUE TITLE
Support using `DEAFULT` in `INSERT/UPDATE/REPLACE ... SET xxx=DEFAULT`  statement

### ISSUE THREAD

**Issue by wjhuang2016** [pingcap/tidb#11939]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
mysql> create table t (a int default 10);
mysql> update t set a=default;
```

2. What did you expect to see?
```
mysql> create table t (a int default 10);
Query OK, 0 rows affected (0.06 sec)

mysql> update t set a=default;
Query OK, 0 rows affected (0.01 sec)
Rows matched: 0  Changed: 0  Warnings: 0
```

3. What did you see instead?
```
mysql> create table t (a int default 10);
Query OK, 0 rows affected (0.00 sec)

mysql> update t set a=default;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 22 near ""
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
TiDB v3.0.3
TiDB v2.1.16

**Comment by Deardrops** [comment]
`Replace` statment also has this problem
```
mysql> create table t (a int default 10);
mysql> replace t set a=default;
```

**Comment by lonng** [comment]
Seems we should change the parser to make this statement legal.

**Comment by tangenta** [comment]
Fixed in pingcap/parser#551.

**Comment by Deardrops** [comment]
```sql
create table t1 (a int default 1, b int default 2);
insert t1 value (10,10);
update t1 set b = default;
```
In TiDB, The last sql statement will panic.
